src/config.py

```diff
@@ -208,11 +303,13 @@ def server_properties(self) -> List[str]:
         Returns:
             List of properties to be set
         """
-        host = (
-            self.charm.model.get_relation(PEER).data[self.charm.unit].get("private-address", None)
-        )
-        port = 9093 if self.charm.tls.enabled else 9092
-        protocol = "SASL_SSL" if self.charm.tls.enabled else "SASL_PLAINTEXT"
+        protocol_map = [self.internal_listener.protocol_map()]
+        listeners_repr = [self.internal_listener.listener()]
+        advertised_listeners = [self.internal_listener.advertised_listener()]
+        for listener in self.extra_listeners:
+            protocol_map.append(listener.protocol_map())
+            listeners_repr.append(listener.listener())
+            advertised_listeners.append(listener.advertised_listener())
```

Comment:
Mentioned in above comment about having a singular `listeners` property, would be nice to have this in 3 lines rather than 7.